### PR TITLE
cli: Implement 'shelley transaction sign' command

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Key.hs [hunk 1]
```diff
@@ -82,8 +82,8 @@ prettyPublicKey vk =
 -- TODO:  we need to support password-protected secrets.
 -- | Read signing key from a file.  Throw an error if the file can't be read or
 -- fails to deserialise.
-readSigningKey :: CardanoEra -> SigningKeyFile -> ExceptT CliError IO SigningKey
-readSigningKey era (SigningKeyFile fp) = do
+readEraSigningKey :: CardanoEra -> SigningKeyFile -> ExceptT CliError IO SigningKey
+readEraSigningKey era (SigningKeyFile fp) = do
```
Comment (intricate) [inline cardano-cli/src/Cardano/CLI/Key.hs:86]: Appreciate the rename :smile:

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 2]
```diff
@@ -641,6 +644,16 @@ pColdSigningKeyFile =
      <> Opt.help "Filepath of the cold signing key."
      )
 
+pSomeSigningKeyFiles :: Parser [SigningKeyFile]
+pSomeSigningKeyFiles =
+  some $
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:649]: :+1:

**Pull request merged**